samples/sample_common/include/sample_utils.h

```diff
@@ -495,6 +495,53 @@ mfxF64 CalculateFrameRate(mfxU32 nFrameRateExtN, mfxU32 nFrameRateExtD);
 mfxU16 GetFreeSurfaceIndex(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
 mfxU16 GetFreeSurface(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
 void FreeSurfacePool(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
+
+class mfxBitstreamWrapper : public mfxBitstream
+{
+public:
+    mfxBitstreamWrapper()
+        : mfxBitstream()
+    {}
+
+    mfxBitstreamWrapper(mfxU32 n_bytes)
+        : mfxBitstream()
+    {
+        Extend(n_bytes);
+    }
+
+    mfxBitstreamWrapper(const mfxBitstreamWrapper & bs_wrapper)
+    {
+        m_data = bs_wrapper.m_data;
+
+        static_cast<mfxBitstream&>(*this) = static_cast<const mfxBitstream&>(bs_wrapper);
+
+        Data = m_data.data();
+    }
+
+    mfxBitstreamWrapper & operator= (mfxBitstreamWrapper bs_wrapper)
+    {
+        std::swap(*this, bs_wrapper);
+
+        return *this;
+    }
+
+    mfxBitstreamWrapper(mfxBitstreamWrapper && bs_wrapper)           = default;
+    ~mfxBitstreamWrapper()                                           = default;
+
+    void Extend(mfxU32 n_bytes)
```

Comment:
Method's name & its implementation implies that we have no way to reduce bitstream size. Do we really need this constrain?

Reply:
I think yes. I just constrained to existing usage. Do you know any real use case, when we need to shrink used bitstream? It might be enhanced to shrink and forget all data before ```DataOffset``` but other ways we will possibly lose bitstream data